src/renderers/shared/fiber/ReactFiberCommitWork.js

```diff
@@ -302,10 +307,22 @@ module.exports = function<T, P, I, TI, C, CX>(
   }
 
   function commitDeletion(current : Fiber) : void {
-    // Recursively delete all host nodes from the parent.
-    const parent = getHostParent(current);
-    // Detach refs and call componentWillUnmount() on the whole subtree.
-    unmountHostComponents(parent, current);
+    // Recursively delete the closest child host nodes from the closest host parent.
+    // Then detach refs and call componentWillUnmount() on the whole subtree.
+    if (current.tag === HostPortal) {
+      // When deleting a portal, there are no host nodes above it.
+      // It is a host parent itself.
```

Comment:
The "can render nested portals" test definitely gets a `commitDeletion()` call from outside `CommitWork` with a portal. So inlining it is not enough.

Alternatively we could figure out how to avoid tagging portals themselves with `Deletion`, and tag their children instead. Not sure if there's any benefit to this?